.github/workflows/documentation.yml

```diff
@@ -3,28 +3,39 @@ on:
   push:
     branches:
       - main
-    paths:
-      - 'doc/**'
-      - 'api/proto/**'
-      - '.github/workflows/documentation.yml'
   pull_request:
-    paths:
-      - 'doc/**'
-      - 'api/proto/**'
-      - '.github/workflows/documentation.yml'
   workflow_dispatch:
   workflow_call:
 
 permissions:
   contents: write
 jobs:
+  documentation_changes:
+    name: Check If Documentation Changed
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      documentation: ${{ steps.filter.outputs.documentation }}
+    steps:
+    - uses: actions/checkout@v3.5.3
+    - uses: dorny/paths-filter@v2.11.1
+      id: filter
+      with:
+        filters: |
+          documentation:
+              - 'doc/**'
+              - 'api/proto/**'
+              - '.github/workflows/documentation.yml'
   deploy:
+    name: Deploy Documentation
+    needs: documentation_changes
+    if: ${{ needs.documentation_changes.outputs.documentation == 'true' || github.ref_name == 'main' || github.ref_type == 'tag' }}
```

Comment:
But that means, that the new path filtering is only active for PRs and not for main, right?

Reply:
Yes the path filtering is active for PRs and skipped for pushes on main and releases. I have kept the conditional simple and weak to reduce complexity. For me it was important that it not runs on every commit in a PR. But if it runs on every push on main, it is not too bad because merging a PR happens quite less often then a committing into a PR.

We can make it stricter like this I think (not tested):
`if: ${{ needs.documentation_changes.outputs.documentation == 'true' || github.ref_name == 'main' && needs.documentation_changes.outputs.documentation == 'true' || github.ref_type == 'tag' }}`

Reply:
OK I see. Then the change like in the original PR is OK for me.